src/components/VmActions/AutoSelectConsole.js

```diff
@@ -0,0 +1,133 @@
+import React, { useEffect, useState } from 'react'
+import { connect } from 'react-redux'
+import PropTypes from 'prop-types'
+import { addUserMessage, downloadVmConsoles } from '_/actions'
+import ConsoleConfirmationModal from '_/components/VmActions/ConsoleConfirmationModal'
+import { msg } from '_/intl'
+
+const AutoSelectConsole = (props) => {
+  const {
+    vm,
+    show,
+    defaultConsole,
+    isNoVNC,
+    fetchConsoles,
+    requestActive,
+    id,
+    onFailure,
+  } = props
+
+  const initialState = {
+    // consoles: [],
+    selectedProtocol: undefined,
+    failureReason: undefined,
+    dispatchStatus: 'doNotDispatch',
+    active: false,
+  }
+
+  const [state, setState] = useState(initialState)
+  const {
+    dispatchStatus,
+    selectedProtocol,
+    active,
+    // failureReason,
+    // consoles,
+  } = state
+
+  useEffect(() => {
+    if (show) {
```

Comment:
Consider data driven design instead of introducing state flags (`active, dispatchStatus, show, ...`).
Basic flow:
1. No data -> fetch data (to the Redux store)
2. there is data -> render data

Check [VmCardList](https://github.com/oVirt/ovirt-web-ui/blob/8a5a26cb2c3310a67d05673d3482bf1a751cc0a9/src/components/VmsList/VmCardList.js#L44)

Reply:
@rszwajko  I'm not sure if it's the same case, in `vmCardList` data fetching should be triggered automatically, but in this case it should be triggered by click, so we must use flags to avoid fetching when it's not needed...

Reply:
Usually in React/Redux a component just renders data from props. If we need more data then we dispatch an action that fetches missing data and saves it in the store. From there the components are notified via props.
Fetch logic (i.e. error handling) is placed in saga-layer.

In case of AutoSelect the rendered piece of data is `selectedProtocol` - we could place it in the store and populate via some kind of `fetchAndSelectConsole` saga. This would remove all error/state handling logic from the component. Perhaps it would even make the component obsolete since it's more a date provider then a regular component.